llvm/lib/Target/CheerpBackend/CheerpTargetTransformInfo.cpp

```diff
@@ -0,0 +1,111 @@
+//===-- CheerpTargetTransformInfo.cpp - Cheerp helper --------------------===//
+//
+//                     Cheerp: The C++ compiler for the Web
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+// Copyright 2022 Leaning Technologies
+//
+//===----------------------------------------------------------------------===//
+
+#include "CheerpTargetTransformInfo.h"
+#include "llvm/CodeGen/CostTable.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Cheerp/CommandLine.h"
+using namespace llvm;
+
+#define DEBUG_TYPE "wasmtti"
+
+unsigned CheerpTTIImpl::getNumberOfRegisters(unsigned ClassID) const {
+  unsigned Result = BaseT::getNumberOfRegisters(ClassID);
+
+  // For SIMD, use at least 16 registers, as a rough guess.
+  bool Vector = (ClassID == 1);
+  if (!WasmSIMD)
+    return 0;
+  if (LinearOutput == LinearOutputTy::AsmJs)
+    return 0;
+  if (Vector)
+    Result = std::max(Result, 16u);
+
+  return Result;
+}
+
+TypeSize CheerpTTIImpl::getRegisterBitWidth(
+    TargetTransformInfo::RegisterKind K) const {
+  switch (K) {
+  case TargetTransformInfo::RGK_Scalar:
+    return TypeSize::getFixed(64);
```

Comment:
shouldn't this be 32?

Reply:
Not sure, I took this to mean the maximum width of a register for normal scalars, which I thought was 64 (doubles and i64)

Reply:
right, and there is no distinction based on different types? then it's ok